modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2291,15 +2291,13 @@ def dropna(self, **kwargs):
 
     def drop(self, index=None, columns=None):
         if index is not None:
-            # The unique here is to avoid duplicating rows with the same name
             index = np.sort(
-                self.index.get_indexer_for(self.index[~self.index.isin(index)].unique())
+                self.index.get_indexer_for(self.index.difference(pandas.Index(index)))
```

Comment:
Why calling of `Index` constructor is needed here?

Reply:
There is no need in `Index` constructor here :) Fixed

Reply:
LGTM! However could you double check perf?

Reply:
The performance numbers were updated (spoiler: there are no any visible changes)

Reply:
Thanks @prutskov!